CareKit/CareKit/iOS/Task/Controllers/OCKSliderLogTaskController.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  OCKSliderLogTaskController.swift
+//  
+//
+//  Created by Dylan Li on 5/26/20.
+//  Copyright © 2020 NetReconLab. All rights reserved.
+//
+
+#if !os(watchOS)
+
+import CareKitStore
+import Combine
+import Foundation
+import SwiftUI
+
+open class OCKSliderLogTaskController: OCKTaskController {
+    
+    /// Data used to create a `CareKitUI.SliderTaskTaskView`.
+    @Published public private(set) var viewModel: SliderLogTaskViewModel? {
+        willSet { objectWillChange.send() }
+    }
+    
+    /// Data used to create a `CareKitUI.SliderTaskTaskView`.
+    @Published public private(set) var value: Double = 0 {
+        willSet { objectWillChange.send() }
+    }
+    
+    private var cancellable: AnyCancellable?
+    
+    public required init(storeManager: OCKSynchronizedStoreManager) {
+        super.init(storeManager: storeManager)
+        cancellable = $taskEvents.sink { [unowned self] taskEvents in
+            self.viewModel = self.makeViewModel(from: taskEvents)
+        }
+    }
+    
+    private func makeViewModel(from taskEvents: OCKTaskEvents) -> SliderLogTaskViewModel? {
+        guard !taskEvents.isEmpty else { return nil }
+
+        if let foundValue = taskEvents.first?.first?.outcome?.values.last?.numberValue?.doubleValue {
+            value = foundValue
```

Comment:
This issue is still popping up. Basically, the value shown on the slider and button is sometimes outdated. It behaves like a threading issue where it will show the previous value when the button is pressed, but not the current.

This can be seen when testing the slider in Catalog. Sometimes all of the sliders will jump to the wrong value.